### ISSUE TITLE
[BUG] Java provider rules not triggering for patterns ending with * or (*)

### ISSUE THREAD

**Issue by joe-braley** [konveyor/analyzer-lsp#736]
[sample1.zip](https://github.com/user-attachments/files/17852711/sample1.zip)
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

v0.6.0-alpha.2

### Priority

Major

### Current Behavior

Hi team!

I have created sample Java files to test analysis against. When running assessments, we noticed rules with patterns ending in * or (*) are not triggering. This results in missing issues in the generated static report.

For the attempt I have run for this issue the command I ran was:
` ./kantra analyze --input ./sample1/ --target azure-appservice --output ./sample1-output --overwrite`

I believe on the sample code, which is attached, [this app-service rule should](https://github.com/konveyor/rulesets/blob/main/default/generated/azure/09-eap-to-azure-appservice-environment-variables.windup.yaml) have triggered and added an incident in the static report. However, I only observe a single rule being triggered with respect to Windows style pathing in the code.

Please let me know if I performed any steps/categorized incorrectly! This is my first issue in this repo 😄 

### Expected Behavior

The rule linked above would trigger and additional incidents would appear in the static report.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. Download the sample Java code attached to this issue.
1. When running an assessment, set the target to `azure-appservice`.
1. Execute the assessment.
1. Observe that the rule "09-eap-to-azure-appservice-environment-variables.windup.yaml" with the following patterns did not return results:
   - java.referenced:
        location: METHOD_CALL
        pattern: java.lang.System.getenv(*)
    - java.referenced:
        location: METHOD_CALL
        pattern: java.lang.System.getProperty(*)

### Environment

```markdown
version: v0.6.0-alpha.2
SHA: 2f22d6db9f4ed5e64ba211af082fd8e9de9b7e01
image: quay.io/konveyor/kantra

PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian

Python 3.11.2
Java 17.0.13+11
```

### Anything else?

The result of the latest analysis on the attached sample I performed using the above Kantra version. Yields only one issue:
![Image](https://github.com/user-attachments/assets/5d2ac929-ad6a-4802-8d1b-f5873627d895)

This view is to show the sample code with the Java function calls. I believe we should see more issues when these are present in the code.
![Image](https://github.com/user-attachments/assets/661986df-b528-42a2-af51-3b932a91b2ba)

**Comment by joe-braley** [comment]
CC @kthatipally

**Comment by jmle** [comment]
I have been taking a quick look at this and it seems that there is a problem when matching method calls in general. We will need to take a deeper look into it.

**Comment by jmle** [comment]
I have discovered some things after investigating. Having a `System.getenv()` call in the code:

- `getenv` and `getenv()` both work.
- `getenv*` also works
- When having methods with argument types (for instance, `rest.getForObject(new URI("http://www.example.com/"), Source.class);`), the following works: `getForObject(*,*)` and `getForObject(URI,Class<Source>)` The following does not: `getForObject(*)`.
- Qualifying the method in the previous example works, ie looking for `org.springframework.web.client.RestTemplate.getForObject`, but having `System.getenv()` in the code and looking for `java.lang.System.getenv` does not.

My conclusions after this:
- There seems to be a problem when trying to look for static method calls in the LS. I am not sure what's causing this. A potential fix would be to change these rules to an `and` condition requiring both the call `getenv` and the class `System` to be used in the same file. More needs to be investigating, maybe trying to contact the Eclipse people.
- The `(*)` pattern only works if there is a single argument to the method, otherwise it doesn't. There must be as many `*` as arguments in the method. Some rules may need a fix.

FYI @rromannissen @dymurray

**Comment by shawn-hurley** [comment]
I wonder if for method searches it makes sense to remove the arguments if there is a `(*)` assuming that the rules want to find all overloaded methods. 

As for the static functions, I do wonder if there is a special parameter that we have to add in the search to consider these?

**Comment by zhoufenqin** [comment]
it seems that the issue still exist, is there any plan to fix it?

**Comment by shawn-hurley** [comment]
I don't know if we have returned to this issue and figured out a "correct" solution. If you would like to take this I can assign you the issue and we can discuss the fix in the issue or a PR if you have ideas. 

WDYT?

**Comment by jmle** [comment]
There are two sides to this issue according to my conclusions from november:
1. Static method calls do not work when used with FQNs. [This issue](https://github.com/konveyor/analyzer-lsp/issues/769) covers that problem.
2. The `(*)` pattern doesn't work when the method has more than one argument. This is being fixed as part of my work to bring the tests from Windup into Konveyor (I am also fixing rules that didn't work).

**Comment by shawn-hurley** [comment]
@jmle I am interested in the work that @pranavgaikwad did to find the actual package with AST parsing, which has fixed this issue. Do you have a test lying around to check, or are you still seeing it pretty regularly?

**Comment by jmle** [comment]
@shawn-hurley this is still happening according to my latest tests

**Comment by haoozhang** [comment]
I have some findings for the following rule conditions,
```
  when:
    or:
    - java.referenced:
        location: METHOD_CALL
        pattern: java.lang.System.getenv*
    - java.referenced:
        location: METHOD_CALL
        pattern: java.lang.System.getProperty*
```

When I run against the following java code file, it cannot match the method call `System.getenv` and `System.getProperty`.
```java
package org.sample;

public class SystemGetEnvGetProperty {

    public static void main(String[] args) {
        int PORT = -1;

        // Attempt to read a port number from the environment
        try {
            PORT = Integer.parseInt(System.getenv("PORT"));
        } catch (Exception ex) {
        }

        // Attempt to read the port number from a System property
        try {
            PORT = Integer.parseInt(System.getProperty("PORT"));
        } catch (Exception ex) {
        }

        if (PORT < 0) {
            PORT = 8080;
        }

        System.out.println(String.format("Found port: %d", PORT));
    }

}
```

But If I explicitly import the package `java.lang.System` on the top of java file, it can match.
```java
package org.sample;

import java.lang.System;

public class SystemGetEnvGetProperty {

// rest of code
```

So, there may be a gap that **the underlying language server cannot automatically include all of the files under `java.lang` package when performing java source code analysis, but Java will help to implicitly include them when we write code (we don't have to manually import it)**.

**Comment by jmle** [comment]
It would be good to check with the LS team if this is something that makes sense to them, what do you think @shawn-hurley
